kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -495,6 +499,56 @@ public static Optional<Tuple2<Protocol, Set<TableFeature>>> autoUpgradeProtocolB
     }
   }
 
+  /**
+   * Adds features from currentProtocol based on overrides in properties.
+   *
+   * <p>Overrides are specified using a key in th form {@linkplain
+   * #PROPERTIES_FEATURE_OVERRIDE_PREFIX} + {featureName}. (e.g. {@code
+   * delta.feature.icebergWriterCompatV1}). The value should be "supported" to add the feature.
+   * Currently, removing values is not handled.
+   *
+   * @return An updated properties map and protocol if there are overrides present and the features
```

Comment:
`updated properties map` --> how is it updated? what has changed about it? i.e. we have removed the table properties that start with delta.feature. ---> those are NOT persisted to the deltalog, right?